src/components/schedule-visualizer/robot-image-icon.tsx

```diff
@@ -16,12 +16,28 @@ const RobotImageIcon = React.forwardRef(function(
   props: RobotImageIconProps,
   ref: React.Ref<SVGGElement>,
 ): React.ReactElement {
-  const { robot, footprint, iconPath, dispatchIconError, inConflict } = props;
+  const { robot, footprint, iconPath, dispatchIconError, inConflict, colorManager } = props;
+
   const theme = useTheme();
   // The default icon uses footprint as the radius, so we * 2 here because the width/height
   // is in a square. With the double size of the footprint, we achieved a similar
   // size to the robot default svg icon.
   const [imgIconWidth, imgIconHeigth] = useMemo(() => [footprint * 2, footprint * 2], [footprint]);
+  const [robotColor, setRobotColor] = useState<string | null>(() =>
+    colorManager.robotColorFromCache(robot.name),
+  );
+
+  React.useLayoutEffect(() => {
+    if (robotColor) {
+      return;
+    }
+    (async () => {
+      await colorManager.robotImageColor(iconPath, robot.name).then(color => {
```

Comment:
Correct me if I am wrong, but what you trying to say is if vibrant fails to extract a color then we can directly load the default icon?

Because I thought the check for the loading of image icons is already done in `robot.tsx`?

Reply:
> Because I thought the check for the loading of image icons is already done in `robot.tsx`?

Mmmhh.. `robots.tsx` checks if a path exists for a fleet or fleet model and it doesn't check for loading success. `robot-image-icon.tsx` is where the image is loaded and the <image> component is the one in charge of notifying us if the load failed `onError=`(due to a bad path or a missing icon). If the load fails, it throws an error and the `robot.tsx ` component is in charge of switching to the default icon.

> Correct me if I am wrong, but what you trying to say is if vibrant fails to extract a color then we can directly load the default icon?

Yeah, if vibrant fails because it wasn't able to get the image. If it fails for something else, we can either set a default color or as you said load directly the default icon.